src/Native/System.IO.Compression.Native/pal_zlib.h

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pal_types.h"
+
+/*
+A structure that holds the input and output values for the zlib functions.
+*/
+struct PAL_ZStream
+{
+    uint8_t* nextIn; // next input byte
+    uint8_t* nextOut; // next output byte should be put there
+
+    uint32_t availIn; // number of bytes available at nextIn
+    uint32_t availOut; // remaining free space at nextOut
```

Comment:
I did move them around from my original order, since I was getting a compiler warning/error.

This was the first time I had to deal with padding. Is it good practice to put all the pointers together and then the integer types? Is there some sort of general ordering practice?